File: leetcode-algorithms/946. Validate Stack Sequences/solution.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> s;
        int i=0;
        for(int p : pushed){
            s.push(p);
            while(s.size() && s.top() == popped[i])
                s.pop(),i += 1;
        }
        return s.empty();
    }
};

File: leetcode-algorithms/946. Validate Stack Sequences/solution.py
def validateStackSequences(pushed, popped):
	"""
	:type pushed: List[int]
	:type popped: List[int]
	:rtype: bool
	"""
	temp = []
	i = 0
	for p in pushed:
		temp.append(p)
		while len(temp)>0 and temp[-1] == popped[i]:
			temp.pop()
			i += 1
	return len(temp) == 0

if __name__ == '__main__':
	pushed = [1,2,3,4,5]
	popped = [4,5,3,2,1]

	res = validateStackSequences(pushed,popped)

	print(res)
			